connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -304,7 +304,7 @@ public void configure(Map<String, ?> configs, boolean isKey) {
 
         Object cacheSizeVal = configs.get(SCHEMAS_CACHE_SIZE_CONFIG);
         if (cacheSizeVal != null)
-            cacheSize = (int) cacheSizeVal;
+            cacheSize = Integer.parseInt((String) cacheSizeVal);
```

Comment:
Hmm, so actually I think the problem here is that we need to be able to parse generally because these can technically be passed in as `Map<String, ?>`, so the type is unknown.

Another option is to replace this parsing with `ConfigDef`, which will handle both the `String` and `Integer` cases since its designed to handle both properties files and `Maps` passed in directly from code. To be honest, I think the only reason we didn't do this before is because this code came with the initial patch that was prototyped outside of Kafka and couldn't use internal APIs.